sdk/resourcemanager/azure-resourcemanager-appplatform/src/main/java/com/azure/resourcemanager/appplatform/implementation/SpringServiceImpl.java

```diff
@@ -214,7 +249,11 @@ public SpringServiceImpl withGitConfig(ConfigServerGitProperty gitConfig) {
 
     @Override
     public SpringServiceImpl withoutGitConfig() {
-        return withGitConfig(null);
+        if (isEnterpriseTier()) {
+            return withGitConfig((ConfigurationServiceGitProperty) null);
+        } else {
+            return withGitConfig((ConfigServerGitProperty) null);
```

Comment:
nit, Do we have to use method overload? Might it be easier if just different name?

Reply:
Agree, I'm considering `withDefaultGitRepository()` and `withGitRepository()`. Will change it in next PR.